pkgs/by-name/so/solana-cli/package.nix

```diff
@@ -102,6 +99,12 @@ rustPlatform.buildRustPackage rec {
       Libsystem
     ];
 
+  doInstallCheck = true;
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgram = "${placeholder "out"}/bin/solana";
+  versionCheckProgramArg = "--version";
```

Comment:
Is this new? I didn't know it was a thing :o

Reply:
Yup! Pretty recent. It should replace `passthru.tests.version`